### ISSUE TITLE
swapAndInvoice only if card can be charged

### ISSUE THREAD

**Issue by fonz-lu** [laravel/cashier#238]
Hi,

where do I need to catch the \Stripe\Error\Card for an existing customer if he swaps the plan? For now it catches the card declined error only once when I do this

``` php
try {
$user->subscription ( $request->id )->prorate ()->swapAndInvoice ();
} catch ( \Stripe\Error\Card $e ) {
...
```

And then in stripe, the user is switched to the new plan, with an invoice and charge. And I don't want that to happen. Afterwards there is no charge declined message, because the prorate is based on a payment due.

What I want is that the user is only switched to the new plan if the card can be charged and not switch to the new one, as it is done for $user->subscription('monthly')->create($creditCardToken);
